src/System.Threading/src/System/Threading/LockCookie.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    public struct LockCookie
+    {
+        internal LockCookieFlags _flags;
+        internal ushort _readerLevel;
+        internal ushort _writerLevel;
+        internal int _threadID;
+
+        public override int GetHashCode()
+        {
+            return (int)_flags + _readerLevel + _writerLevel + _threadID;
```

Comment:
Not a particularly good hash code :)  But then again, this matches what's in desktop, RWLS should be used in general instead of RWL, and if you're storing LockCookies in a dictionary something is likely quite wrong :)